src/main/scala/org/allenai/plugins/DeployPlugin.scala

```diff
@@ -245,6 +245,37 @@ object DeployPlugin extends AutoPlugin {
     log.info("Deploy complete. Validate your server!")
   }
 
+  // we stage and generate the cache key in the same task so that stage only runs once
+  val stageAndCacheKeyTask = stageAndCacheKey := {
+    import VersionInjectorPlugin.autoImport.{ gitLocalSha1 }
+    val stageDir = (UniversalPlugin.autoImport.stage in thisProject).value
+    val logger = streams.value.log
+    val allFiles = ((stageDir / "bin" +++ stageDir / "conf" +++ stageDir / "lib" +++ stageDir / "public") * ".jar").get
```

Comment:
You're filtering for jars, which we only store in `lib`. I think that's the only directory we want here - other jars won't even end up on our classpath . . .